### PR TITLE
api: fix the certificate problem in API client

### PR DESCRIPTION
**Pull request by nolouch**
Signed-off-by: nolouch <nolouch@gmail.com>

### What problem does this PR solve? 
Fix https://github.com/pingcap/pd/issues/2362

### What is changed and how it works?
add TLS supports for the client.

### Release note 
- Fix the certificate problem in API client

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)

### CODE REVIEW

File: server/api/util.go [hunk 1]
```diff
@@ -20,17 +20,14 @@ import (
...
-var dialClient = &http.Client{
-	Transport: &http.Transport{
-		DisableKeepAlives: true,
-	},
-}
-
 var (
+
+	// dialClient used to dial http request.
+	dialClient        = cluster.DialClient
```
Comment (rleungx) [inline server/api/util.go:30]: The client in cluster package has a timeout. Does the client in API needs it?

Comment (nolouch) [reply server/api/util.go:30]: I checked it and it only used in a redirect for the scheduler handler, I think it's ok to timeout.

Comment (rleungx) [reply server/api/util.go:30]: If we use API to get all regions and the region number is large, will it break the process of getting regions?

Comment (nolouch) [reply server/api/util.go:30]: the client in the user, not in PD.

Comment (rleungx) [reply server/api/util.go:30]: got

**Pull request merged**